flink-optimizer/src/main/java/org/apache/flink/optimizer/dag/PartitionNode.java

```diff
@@ -90,13 +90,20 @@ public SemanticProperties getSemanticProperties() {
 		private final PartitionMethod pMethod;
 		private final Partitioner<?> customPartitioner;
 		private final DataDistribution distribution;
-		
+		private final Ordering ordering;
+
 		public PartitionDescriptor(PartitionMethod pMethod, FieldSet pKeys, Partitioner<?> customPartitioner, DataDistribution distribution) {
```

Comment:
Can you remove this constructor? I think the optimizer will fail if no `ordering` is set.